py/terra_widgets/html_snapshots.py

```diff
@@ -0,0 +1,371 @@
+"""Methods for creating and viewing HTML snapshots of notebooks.
+
+It is useful to create and retain HTML snapshots of notebooks to track how results within
+a notebook changes over time as analysis work proceeds.
+"""
+
+import collections
+import os
+import tempfile
+from typing import Dict
+from typing import List
+
+from IPython import get_ipython
+from IPython.display import display
+from IPython.display import HTML
+from IPython.display import IFrame
+from ipywidgets import widgets
+import pandas as pd
+import tensorflow as tf
+
+from terra_widgets.workspace_metadata import WorkspaceMetadata
+from terra_widgets.workspace_paths import WorkspacePaths
+
+# Define this in the outer scope so that it lives for the duration of the Jupyter kernel.
+TEMP_HTML = tempfile.NamedTemporaryFile(dir=os.getcwd(), prefix='view_an_html_snapshot_', suffix='.html')
+
+# Configure notebook display preferences to better suit this UI.
+if pd.__version__.startswith('1'):
+  pd.set_option('display.max_colwidth', None)
+else:
+  pd.set_option('display.max_colwidth', -1)
+get_ipython().run_cell_magic(
```

Comment:
Hey Nicole - I missed this on my first review. Will this affect other output cells that get created by the user within the notebook? Also - thoughts on moving this out of the global context? When I tried importing this library just to test whether it could be imported, it failed trying to execute ipython magic (I wasn't in a notebook context). That's not a major issue, but I think changing the display might be too surprising of a side effect for importing the package. Maybe running this lazily when the user renders a widget would be better.

Reply:
Yes, it affects the display of all cells run thereafter. It's no problem to move it out of the global scope. I'll send a PR.

Note that module `html_snapshots` only has methods relevant to notebooks - it's not useful in other contexts. The other two modules are not specific to notebooks.

If we wanted to have it work in other contexts, such as embedding it on an AoU workbench page, it looks like the controls would work https://ipywidgets.readthedocs.io/en/latest/embedding.html but we would need to revisit the implementation of the Cloud Storage interactions.